kratos/includes/define.h

```diff
@@ -24,6 +24,7 @@
 #include "includes/kratos_export_api.h"
 #include "includes/shared_pointers.h"
 #include "includes/exception.h"
+#include "includes/kratos_version.h"
```

Comment:
BTW I also just realized that including this will basically require a recompilation of the entire Kratos each time a commit is made due to the change or the version Hash!
I am not blocking since not my call but I think this unfortunately is a serious disadvantage!

Reply:
Well, an option will be to move the definition of the major/minor and version number to an independent file.

Reply:
ouch. i do see this as a blocker!!

also this does not need to be in the define.h isn't it? eventually we could have a define_deprecation.h

Reply:
Ok

Reply:
Done, but this doesn't solve the commit has problem at 100% (element.h and condition.h use the macro, so many files are affected). @roigcarlo can we split the kratos_version.h in two halves?, one with the version number, major and minor patch and the second which includes this one with the hash (using the same name so no more code modification is needed)

Reply:
now it will still recompile most of Kratos ...

splitting the version could work I think

Reply:
Yep we can split it, no problem for me.

Reply:
Actually we have another PR open in this direction by @philbucher. Give me 10 min to see if I can find another solution for the version thing.

Reply:
See #4741